### ISSUE TITLE
Script update-yaml.py fails to re-create base configuration file with overrides

### ISSUE THREAD

**Issue by mrhillsman** [rcbops/rpc-openstack#950]
##### What version (commit) do you encounter this issue on?

r12.0.0

##### What type of issue is this (bug, feature, documentation, regression)?

Bug

##### What is the expected behavior?

Merging of base configuration file ```/etc/openstack_deploy/user_variables.yml``` with overrides file ```/opt/rpc-openstack/rpcd/etc/openstack_deploy/user_variables.yml``` using python script ```/opt/rpc-openstack/scripts/update-yaml.py``` into a new file ```/etc/openstack_deploy/user_variables.yml``` which contains the content of both the base configuration and the overrides file.

##### What is actually happening?

An error occurs:

```
root@infra01:/opt/rpc-openstack# /opt/rpc-openstack/scripts/update-yaml.py /etc/openstack_deploy/user_variables.yml /opt/rpc-openstack/rpcd/etc/openstack_deploy/user_variables.yml 
Traceback (most recent call last):
  File "/opt/rpc-openstack/scripts/update-yaml.py", line 52, in <module>
    config = dict(base.items() + overrides.items())
AttributeError: 'NoneType' object has no attribute 'items'
```

##### How can we reproduce this issue?

SSH into deployment host
```
# cd /opt
# git clone -b r12.0.0 --recursive http://github.com/rcbops/rpc-openstack
# cd rpc-openstack/openstack-ansible
# cp -r etc/openstack_deploy /etc/
# scripts/bootstrap-ansible.sh
# cd /opt/rpc-openstack
# scripts/update-yaml.py /etc/openstack_deploy/user_variables.yml /opt/rpc-openstack/rpcd/etc/openstack_deploy/user_variables.yml
```
##### How can we verify your issue is addressed?

Without modifying the base configuration file ```/etc/openstack_deploy/user_variables.yml``` the file ```/opt/rpc-openstack/scripts/update-yaml.py``` successfully creates ```/etc/openstack_deploy/user_variables.yml``` with the base configuration and overrides in a new ```/etc/openstack_deploy/user_variables.yml``` file

##### If this is a large issue, what can be done immediately to alleviate the issue while working on a longer-term solution?
```shell
diff --git a/scripts/update-yaml.py b/scripts/update-yaml.py
index ad47d50..19eb17c 100755
--- a/scripts/update-yaml.py
+++ b/scripts/update-yaml.py
@@ -32,6 +32,7 @@ def get_config(path):
     try:
         with open(path) as f:
             data = f.read()
+            data = yaml.safe_load(data)
     except IOError as e:
         if e.errno == errno.ENOENT:
             data = None
@@ -42,7 +43,7 @@ def get_config(path):
         return {}
     else:
         # assume config is a dict
-        return yaml.safe_load(data)
+        return data
```
##### What is your evaluation on the importance of this issue, and why?

I would say it is high importance as this issue requires modifying the base configuration and the provided overrides file is not sufficient to resolve.

##### What branches / milestones do you believe this issue should target?
The list below is what is currently supported for backports, remove branches you do not wish to target.
- [x] Master https://github.com/rcbops/rpc-openstack/pull/1198
- [x] liberty-12.2 #1203 
- [ ] ~~liberty-12.1 #1206~~

**Comment by d34dh0r53** [comment]
@codebauss can you provide us with examples of your user_variables files?  I think you may be missing something in one of them.

**Comment by jaime-gutierrez** [comment]
@d34dh0r53 In attempting to install r12.0.0 today, I've also run into the same error as @codebauss with the scripts/update-yaml.py. The user_variables.yml file, /etc/openstack_deploy/user_variables.yml, is completely commented out. 

Here are examples of my user_variables files:

/etc/openstack_deploy/user_variables.yml:
https://gist.github.rackspace.com/jaime-gutierrez/38603ec140111e00b335

/opt/rpc-openstack/rpcd/etc/openstack_deploy/user_variables.yml:
https://gist.github.rackspace.com/jaime-gutierrez/bb67be6f3ba9bc02f2eb

**Comment by shane-c** [comment]
I think the problem is the following commit 

https://github.com/openstack/openstack-ansible/commit/4dadec6178de3392d51366b960ecdfb1b60bdbb5#diff-3b3a638ddf7c68bc4ff9470d533980a0

We can work around this easily if nothing in the OSA version of user_variables.yml is really needed.

**Comment by codebauss** [comment]
Hey @d34dh0r53 
I am using the default files without any modifications to them. The 'steps to reproduce' will get you to the error straightway. I have since deleted the lab environment but can spin up another quite easily if needed.

**Comment by mrhillsman** [comment]
@d34dh0r53 @prometheanfire is there any additional information I can provide outside of the steps to replicate being that no changes were made to the code?

Is this a better option to fix?

diff --git a/scripts/update-yaml.py b/scripts/update-yaml.py
index ad47d50..d1d60fb 100755
--- a/scripts/update-yaml.py
+++ b/scripts/update-yaml.py
@@ -48,6 +48,8 @@ def get_config(path):
 if __name__ == '__main__':
     args = parse_args()
     base = get_config(args.base)
+    if base is None:
+        base = {}
     overrides = get_config(args.overrides)
     config = dict(base.items() + overrides.items())
     if config:

**Comment by txgunslinger77** [comment]
Just ran into this too on a virgin Liberty r12.1.0 install. No files were modified before running the script, and this is on Step 2 of the install docs in the 'Deployment Configuration' section under 'Prerequisites'.

Would be good to know if mrhillsman's fix is valid and how to implement it.

**Comment by mancdaz** [comment]
Hi @mrhillsman - I'm going to put a PR in using your fix (the first one), unless you'd like to do so yourself? Because I think that's a valid bug in the script.

On a separate note, can you point me to the documentation that shows the steps you laid out above? I think we maybe need to tweak the docs because that update_yaml script was never intended to be used in greenfield deployments.

**Comment by sigmavirus24** [comment]
Backported to liberty-12.2 in #1203

**Comment by byronmccollum** [comment]
Would appreciate a liberty-12.1 backport...

**Comment by sigmavirus24** [comment]
Added to the list @byronmccollum 

Will send the PR soon

**Comment by sigmavirus24** [comment]
@byronmccollum #1206 is a 12.1 backport for you.

**Comment by sigmavirus24** [comment]
@byronmccollum and @git-harry just closed #1206 and deleted liberty-12.1.
